aws-greengrass/aws-greengrass-nucleus
# Fix for race condition when service transition from installed to running

**Issue #, if available:**

**Description of changes:**
1. removed unused service from config.yaml
2. Fixed race condition in run lifecycle stage

Race condition:
Thread 1
service executes the run script using thread 2 and sets the stage of the service as running. 
Thread 2
Once run script finishes the service sets its stage as finished. 

If thread 2 sets stage as finished before thread 1 can set the stage running, then thread 2's set stage as finished is over written and the service stays in running forever (if notime out is specified)

**Why is this change necessary:**

Fix race condition and make tests less flaky.

Above race condition causes the below test to fail randomly
KernelTest#GIVEN_expected_stdout_patterns_WHEN_kernel_launches_THEN_all_expected_patterns_are_seen

**How was this change tested:**
Manual testing:
replicated the race condition manually by adding thread.sleep and tested the solution

ran mvn verify 50 times and builds succeeded.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
